sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/QueryPlan.scala

```diff
@@ -699,6 +701,23 @@ abstract class QueryPlan[PlanType <: QueryPlan[PlanType]]
    * significant memory overhead on the driver.
    */
   def allAttributes: AttributeSeq = children.flatMap(_.output)
+
+  final override def validateNodePatterns(): Unit = {
```

Comment:
It seems that this method is never called

Reply:
Also, shall we add a similar check in Expressions?

Reply:
it's called in https://github.com/apache/spark/pull/50805/files#diff-94575875fbf007fdaf43e4946c69c18649294fed974a46816ab1986f6350541bR98

Reply:
@cloud-fan it was a great catch! The call site was added in a new commit.